_books/ion-1-1/src/macros/defining_macros.md

```diff
@@ -165,8 +165,259 @@ Syntactically, the signature is an s-expression of [parameter declarations](#mac
 | `y`  |  `float16`  | `zero-or-one`  |
 | `z`  |  `tagged`   | `one-or-more`  |
 
-<!-- TODO; grammar and examples -->
-
 ### Template definition language (TDL)
 
-<!-- TODO -->
\ No newline at end of file
+The macro's _template_ is a single Ion value that defines how a reader should expand invovations of the macro.
+Ion 1.1 introduces a template definition language (TDL) to express this process in terms of the macro's parameters.
+TDL is a small language with only a few constructs.
+
+A TDL _expression_ can be any of the following:
+1. A literal [Ion scalar](#ion-scalars)
+2. A macro invocation
+3. A variable expansion
+4. A quasi-literal Ion container
+5. A [special form](special_forms.md)
+
+In terms of its encoding, TDL is "just Ion."
+As you shall see in the following sections, the constructs it introduces are written as s-expressions with a distinguishing leading value or values.
+
+A [grammar](#tdl-grammar) for TDL can be found at the end of this chapter.
+
+#### Ion scalars
+
+Ion scalars are interpreted literally. These include values of any type except `list`, `sexp`, and `struct`.
+`null` values of any type—even `null.list`, `null.sexp`, and `null.struct`—are also interpreted literally.
+
+##### Examples
+These macros are constants; they take no parameters.
+When they are invoked, they expand to a stream of a single value: the Ion scalar acting as the template expression.
+```ion
+$ion_encoding::(
+  (macro_table
+    (macro greeting () "hello")
+    (macro birthday () 1996-10-11)
+    // Annotations are also literal
+    (macro price () USD::29.95)
+  )
+)
+
+(:greeting) => "hello"
+(:birthday) => 1996-10-11
+(:price)    => USD::29.95
+```
+
+#### Macro invocations
+
+Macro invocations call an existing macro.
+The invoked macro could be a [system macro](system_macros.md), a macro imported from a [shared module](../todo.md), or a macro previously defined in the current scope.
```

Comment:
Is "scope" equivalent in meaning to "context" in these docs? I think I've seen both, and I'm wondering if we want to be consistent.

Reply:
"Context" is a bit overloaded in that it can mean the encoding context or the more general concept of a context. I used 'scope' here because it's talking about the region of the module definition where the macro is addressable.